src/hooks/useConfigurePatch.ts

```diff
@@ -115,6 +115,15 @@ const tabToIndexMap = {
   [PatchTab.Parameters]: 2,
 };
 
+// Helper to extract type T from an array of type T
+type Unpacked<T> = T extends (infer U)[] ? U : T;
```

Comment:
Can we export this from a util file? There are plenty of places in our codebase where we should use this helper instead of taking the zero index of an array type